src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -715,6 +715,9 @@ fun processLabelResolveVariants(label: RsLabel, processor: RsResolveProcessor):
         if (scope is RsLabeledExpression) {
             val labelDecl = scope.labelDecl ?: continue
             if (processor(labelDecl)) return true
+
+            // Handle the possible label shadowing in nested scopes.
+            if (labelDecl.name == label.text) return true
```

Comment:
It seems we usually do it via "shadowing" resolve processor to make `processor(labelDecl)` return `true`

Reply:
1. You broke completion by this line. Instead, you should likely use `processWithShadowingAndUpdateScope` (or do the thing manually if `processWithShadowingAndUpdateScope` is not applicable in this case)
2. You must *never* use `PsiElement.getText()` in name resolution code because it is always a bug
3. You must never use `return true` to abort the name resolution procedure because `true` signals that it has been returned from the processor, i.e. processor signals by `return true` that it found something it seeking for. In short, you should use `return false` in order to finish the name resolution. But in this case it is harmful anyway because you need to continue walking scopes in order to collect all possible completion variants

Btw using `label.ancestors` is also an (old) bug, so this code doesn't work if a lifetime is used inside a macro call, but the lifetime is declared outside of the macro